nemo/collections/nlp/modules/common/text_generation_utils.py

```diff
@@ -181,14 +181,16 @@ def megatron_neva_generate(model, prompt_dict_list, length_params, sampling_para
 
         clean_response = clean_text
 
-        if conv_template == "nvgpt":
+        if conv_template in ["nvgpt", "nv_steerlm"]:
             labels_str_regexp = re.compile(f"<extra_id_2>quality:.*\n")
             last_match_end_position = None
             for match in re.finditer(labels_str_regexp, clean_response):
                 last_match_end_position = match.end()
             if last_match_end_position is not None:
                 clean_response = clean_response[last_match_end_position:]
             clean_response = clean_response.strip("<extra_id_1>")
+        elif conv_template == 'nv_dpo':
+            clean_response = clean_response.split("<extra_id_1>")[-2].strip().split("\n")[-1]
```

Comment:
`.strip().split("\n")[-1]` need all these? there isa strip on line 199 below

Reply:
`clean_response.split("<extra_id_1>")[-2]` would give "Assistant\n{GENERATION}\n", then use `.strip().split("\n")[-1]` to grab the {GENERATION}. If directly use line 199 "Assistant" remains if I understand correctly